File: libs/models/src/role.ts
import { Column, Entity, JoinTable, ManyToMany, PrimaryGeneratedColumn } from "typeorm";
import { Permission } from "./permission";

@Entity('role')
export class Role {
    @PrimaryGeneratedColumn()
    id: number;
    @Column()
    name: string;
    @ManyToMany(()=>Permission)
    @JoinTable({name: 'role_permission'})
    permission: Permission[]
}

File: tiny-pro/model/src/main/java/com/huaweicloud/model/Role.java
package com.huaweicloud.model;

import jakarta.persistence.*;

import java.util.Set;

@Entity(name="role")
public class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public int id;
    @Column()
    public String name;
    @ManyToMany(targetEntity = Permission.class)
    @JoinTable(name = "role_permission")
    public Set<Permission> permission;
    @ManyToMany(targetEntity = Menu.class)
    @JoinTable(name = "role_menu")
    public Set<Menu> menu;
}
